manager/integration/tests/test_stress.py

```diff
@@ -0,0 +1,316 @@
+        body=pv_manifest
+    )
+
+    pv_name = pv_manifest['metadata']['name']
+
+    pv_status_is_available = False
+    for i in range(RETRY_COUNTS):
+        pv = k8s_api_client.read_persistent_volume_status(
+            name=pv_name
+        )
+
+        if pv.status.phase == "Available":
+            pv_status_is_available = True
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert pv_status_is_available
+
+
+def create_and_wait_pvc(k8s_api_client, pvc_manifest):
+    k8s_api_client.create_namespaced_persistent_volume_claim(
+        namespace="default",
+        body=pvc_manifest
+    )
+    pvc_name = pvc_manifest['metadata']['name']
+
+    pvc_status_is_bound = False
+    for i in range(RETRY_COUNTS):
+        pvc = k8s_api_client.read_namespaced_persistent_volume_claim(
+            namespace="default",
+            name=pvc_name
+        )
+
+        if pvc.status.phase == "Bound":
+            pvc_status_is_bound = True
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert pvc_status_is_bound
+
+
+def write_data(k8s_api_client, pod_name):
+    test_data = generate_random_data(TEST_DATA_BYTES)
+    write_pod_volume_data(k8s_api_client, pod_name, test_data)
+    volume_data = read_volume_data(k8s_api_client, pod_name)
+    assert test_data == volume_data
+
+
+def delete_data(k8s_api_client, pod_name):
+    test_data = generate_random_data(0)
+    write_pod_volume_data(k8s_api_client, pod_name, test_data)
+    volume_data = read_volume_data(k8s_api_client, pod_name)
+    assert volume_data == ""
+
+
+def create_snapshot(longhorn_api_client, volume_name):
+    volume = longhorn_api_client.by_id_volume(volume_name)
+
+    snapshots = volume.snapshotList(volume=volume_name)
+    snapshots_count = len(snapshots.data)
+
+    volume.snapshotCreate()
+
+    snapshots = volume.snapshotList(volume=volume_name)
+
+    assert len(snapshots) == snapshots_count + 1
+
+
+def purge_random_snapshot(longhorn_api_client, volume_name, snapshot_name):
+
+    volume = longhorn_api_client.by_id_volume(volume_name)
+
+    volume.snapshotPurge()
+
+    wait_for_snapshot_purge(
+        longhorn_api_client,
+        volume_name,
+        snapshot_name
+    )
+
+
+def delete_random_snapshot(longhorn_api_client, volume_name):
+    volume = longhorn_api_client.by_id_volume(volume_name)
+
+    snapshots = volume.snapshotList(volume=volume_name)
+
+    snapshots_count = len(snapshots.data)
+
+    while True:
+        snapshot_id = randrange(0, snapshots_count)
+
+        if snapshots.data[snapshot_id].name == "volume-head":
+            continue
+        else:
+            break
+    snapshot_name = snapshots.data[snapshot_id].name
+
+    volume.snapshotDelete(name=snapshot_name)
+
+    trigger_purge = randrange(0, 2)
+
+    if trigger_purge == 1:
+        purge_random_snapshot(longhorn_api_client, volume_name, snapshot_name)
+
+
+def test_stress(generate_load):
+    pass
+
+
+@pytest.fixture(params=count)
+def generate_load(request):
```

Comment:
Will this fixture function be run sequentially? It means we are not operating multiple volumes simultaneously.

Reply:
No, This is actually a workaround to run the test in parallel.
I've added `pytest-xdist` to `requirements.txt` for this reason.
running the stress test in parallel will be done using this command: 
`py.test -n NUM -s test_stress.py`, where NUM is the number of threads that you want to start in parallel.

Reply:
- When you set `NPODS` to a number ,  count is created as a list of integers from `0` to `NPODS-1`.
- `generate_load` is a parameterized fixture with `params=count`, that means that this function will be executed for every value in count list.
- The value of count is passed to generate_load  in `request.param` which i use as an suffix for Volume/PV/PVC/POD names to ensure no duplicate names.
- with `pytest-xdist` we're able to run all these tests together to create a stress test.